services/core/users/src/main/java/org/lfenergy/operatorfabric/users/controllers/EntitiesController.java

```diff
@@ -0,0 +1,215 @@
+        }
+        return null;
+    }
+
+    @Override
+    public Void deleteEntityUser(HttpServletRequest request, HttpServletResponse response, String id, String login) throws Exception {
+
+        //Only existing entities can be updated
+        findEntityOrThrow(id);
+
+        //Retrieve users from repository for users list, throwing an error if a login is not found
+        UserData foundUser = userRepository.findById(login).orElseThrow(()->new ApiErrorException(
+                ApiError.builder()
+                        .status(HttpStatus.NOT_FOUND)
+                        .message(String.format(USER_NOT_FOUND_MSG, login))
+                        .build()
+        ));
+
+        if(foundUser!=null) {
+                foundUser.deleteEntity(id);
+                publisher.publishEvent(new UpdatedUserEvent(this, busServiceMatcher.getServiceId(), foundUser.getLogin()));
+            userRepository.save(foundUser);
+        }
+        return null;
+    }
+
+    @Override
+    public List<? extends Entity> fetchEntities(HttpServletRequest request, HttpServletResponse response) throws Exception {
+        return entityRepository.findAll();
+    }
+
+    @Override
+    public Entity fetchEntity(HttpServletRequest request, HttpServletResponse response, String id) throws Exception {
+        return entityRepository.findById(id).orElseThrow(
+           ()-> new ApiErrorException(
+              ApiError.builder()
+                 .status(HttpStatus.NOT_FOUND)
+                 .message(String.format(ENTITY_NOT_FOUND_MSG, id))
+                 .build()
+           )
+        );
+    }
+
+    @Override
+    public Entity updateEntity(HttpServletRequest request, HttpServletResponse response, String id, Entity entity) throws Exception {
+        //id from entity body parameter should match id path parameter
+        if(!entity.getId().equals(id)){
+            throw new ApiErrorException(
+                    ApiError.builder()
+                            .status(HttpStatus.BAD_REQUEST)
+                            .message(NO_MATCHING_ENTITY_ID_MSG)
+                            .build());
+        } else {
+            return createEntity(request, response, entity);
+        }
+
+    }
+
+    @Override
+    public Void updateEntityUsers(HttpServletRequest request, HttpServletResponse response, String id, List<String> users) throws Exception {
+
+        //Only existing entities can be updated
+        findEntityOrThrow(id);
+
+        List<UserData> formerlyBelongs = userRepository.findByEntitiesContaining(id);
+        List<String> newUsersInEntity = new ArrayList<>(users);
+
+        //Make sure the intended updated users list only contains logins existing in the repository, throwing an error if this is not the case
+        retrieveUsers(users);
+
+        List<UserData> toUpdate = new ArrayList<>();
+        formerlyBelongs.stream()
+           .filter(u->!users.contains(u.getLogin()))
+           .forEach(u-> {
```

Comment:
It could may be possible to use `peek` method to putt all the actions perform upon the userData to update into one place and then collect it in a list in order to save it ? it would more compact and more functional.

Reply:
Code modified following your advice. Thank you Ronan.